components/tailscale/component.go

```diff
@@ -1,40 +1,79 @@
-// Package tailscale tracks the tailscale state (e.g., version) if available.
+// Package tailscale tracks the current tailscale status.
 package tailscale
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
 	"time"
 
 	"github.com/leptonai/gpud/components"
-	tailscale_id "github.com/leptonai/gpud/components/tailscale/id"
 	"github.com/leptonai/gpud/pkg/log"
+	"github.com/leptonai/gpud/pkg/systemd"
 )
 
-func New(ctx context.Context, cfg Config) components.Component {
-	cfg.Query.SetDefaultsIfNotSet()
-
-	return &component{
-		rootCtx: ctx,
-	}
-}
+// Name is the ID of the tailscale component.
+const Name = "tailscale"
 
 var _ components.Component = &component{}
 
 type component struct {
-	rootCtx context.Context
+	ctx    context.Context
+	cancel context.CancelFunc
+
+	checkDependencyInstalled func() bool
+
+	// return true if the systemd service is active, otherwise false
+	checkServiceActive func(context.Context) (bool, error)
+
+	lastMu   sync.RWMutex
+	lastData *Data
 }
 
-func (c *component) Name() string { return tailscale_id.Name }
+func New(ctx context.Context) components.Component {
+	cctx, cancel := context.WithCancel(ctx)
+	c := &component{
+		ctx:                      cctx,
+		cancel:                   cancel,
+		checkDependencyInstalled: checkTailscaledInstalled,
+		checkServiceActive: func(ctx context.Context) (bool, error) {
+			conn, err := systemd.NewDbusConn(ctx)
+			if err != nil {
+				return false, err
+			}
+			defer conn.Close()
+			return conn.IsActive(ctx, "tailscaled")
+		},
+	}
+	return c
+}
+
+func (c *component) Name() string { return Name }
+
+func (c *component) Start() error {
```

Comment:
I didn't find where it was referenced..

Reply:
https://github.com/leptonai/gpud/blob/760ef3b77506ae493e47765aaf3da0ac27c0e4a1/internal/server/server.go#L1247-L1252

Reply:
loll... Overlooked